packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -394,10 +418,16 @@ exports.createNodesForContentType = ({
               // is empty due to links to missing entities
               const resolvableEntryItemFieldValue = entryItemFieldValue
                 .filter(function (v) {
-                  return resolvable.has(v.sys.id)
+                  return resolvable.has(
+                    `${v.sys.id}___${v.sys.linkType || v.sys.type}`
+                  )
                 })
                 .map(function (v) {
-                  return mId(space.sys.id, v.sys.id)
+                  return mId(
+                    space.sys.id,
+                    v.sys.id,
+                    v.sys.linkType || v.sys.type
+                  )
```

Comment:
This sort of thing happens a lot. I think some can be abstracted a little better (but not all).